src/bun.js/bindings/InternalForTesting.cpp

```diff
@@ -0,0 +1,19 @@
+#include "root.h"
+
+#include "ZigGlobalObject.h"
+#include "JavaScriptCore/JSCJSValue.h"
+#include "JavaScriptCore/JSCast.h"
+#include "JavaScriptCore/JSArrayBufferView.h"
+
+namespace Bun {
+
+using namespace JSC;
+
+JSC_DEFINE_HOST_FUNCTION(jsFunction_arrayBufferViewHasBuffer, (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    auto value = callFrame->argument(0);
+    auto view = jsCast<WebCore::JSArrayBufferView*>(value);
```

Comment:
an argument to be made that this could be dynamicCast + RELEASE_ASSERT(view) so that the invariant is for sure caught in ci